show/main.py

```diff
@@ -2916,5 +2916,288 @@ def autorestart(container_name):
 
+#
+# 'vnet' command ("show vnet")
+#
+@cli.group(cls=AliasedGroup, default_if_no_args=False)
+def vnet():
+    """Show vnet related information"""
+    pass
+
+@vnet.command()
+@click.argument('vnet_name', required=True)
+def name(vnet_name):
+    """Show vnet name <vnet name> information"""
+    config_db = ConfigDBConnector()
+    config_db.connect()
+    header = ['vnet name', 'vxlan tunnel', 'vni', 'peer list']
+
+    # Fetching data from config_db for VNET
+    vnet_data = config_db.get_entry('VNET', vnet_name)
+
+    def tablelize(vnet_key, vnet_data):
+        table = []
+        if vnet_data:
+            r = []
+            r.append(vnet_key)
+            r.append(vnet_data.get('vxlan_tunnel'))
+            r.append(vnet_data.get('vni'))
+            r.append(vnet_data.get('peer_list'))
+            table.append(r)
+        return table
+
+    click.echo(tabulate(tablelize(vnet_name, vnet_data), header))
+
+@vnet.command()
+def brief():
+    """Show vnet brief information"""
+    config_db = ConfigDBConnector()
+    config_db.connect()
+    header = ['vnet name', 'vxlan tunnel', 'vni', 'peer list']
+
+    # Fetching data from config_db for VNET
+    vnet_data = config_db.get_table('VNET')
+    vnet_keys = natsorted(vnet_data.keys())
+
+    def tablelize(vnet_keys, vnet_data):
+        table = []
+        for k in vnet_keys:
+            r = []
+            r.append(k)
+            r.append(vnet_data[k].get('vxlan_tunnel'))
+            r.append(vnet_data[k].get('vni'))
+            r.append(vnet_data[k].get('peer_list'))
+            table.append(r)
+        return table
+
+    click.echo(tabulate(tablelize(vnet_keys, vnet_data), header))
+
+@vnet.command()
+def interfaces():
+    """Show vnet interfaces information"""
+    config_db = ConfigDBConnector()
+    config_db.connect()
+
+    header = ['vnet name', 'interfaces']
+
+    # Fetching data from config_db for interfaces
+    intfs_data = config_db.get_table("INTERFACE")
+    vlan_intfs_data = config_db.get_table("VLAN_INTERFACE")
+
+    vnet_intfs = {}
+    for k, v in intfs_data.items():
+        if 'vnet_name' in v:
+            vnet_name = v['vnet_name']
+            if vnet_name in vnet_intfs:
+                vnet_intfs[vnet_name].append(k)
+            else:
+                vnet_intfs[vnet_name] = [k]
+
+    for k, v in vlan_intfs_data.items():
+        if 'vnet_name' in v:
+            vnet_name = v['vnet_name']
+            if vnet_name in vnet_intfs:
+                vnet_intfs[vnet_name].append(k)
+            else:
+                vnet_intfs[vnet_name] = [k]
+
+    table = []
+    for k, v in vnet_intfs.items():
+        r = []
+        r.append(k)
+        r.append(",".join(natsorted(v)))
+        table.append(r)
+
+    click.echo(tabulate(table, header))
+
+@vnet.command()
+def neighbors():
+    """Show vnet neighbors information"""
+    config_db = ConfigDBConnector()
+    config_db.connect()
+
+    header = ['<vnet_name>', 'neighbor', 'interfaces']
```

Comment:
Can you also provide the` mac address `if its available in the attribute?

Reply:
added.
```
admin@ASW-7005:~$ show vnet neighbors 
Vnet_2000    neighbor     mac_address    interfaces
-----------  -----------  -------------  ------------
             11.11.11.11                 Ethernet1


Vnet_3000    neighbor     mac_address        interfaces
-----------  -----------  -----------------  ------------
             20.20.20.20  aa:bb:cc:dd:ee:ff  Vlan2000


```